File: Hour 6/Q2/hands_of_the_clock.java
import java.util.Scanner;

public class hands_of_the_clock{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        try{
            String time = sc.nextLine();
            int h, m;
            float h_angle, m_angle, angle;
            String[] t = time.split(":");
            h = (int)((t[0].charAt(0)-'0')*10 + t[0].charAt(1) - '0');
            m = (int)((t[1].charAt(0)-'0')*10 + t[1].charAt(1) - '0');
            h_angle = 0.5f * (60 * h + m);
            m_angle = 6 * m;
            angle = Math.abs(h_angle - m_angle);
            angle = Math.min(360 - angle, angle);
            System.out.println(angle);
        }
        finally{
            sc.close();
        }
    }
}

File: Hour 6/Q2/hands_of_the_clock.cpp
#include<iostream>
using namespace std;

int main(){
    string time;
    cin>>time;
    int h, m;
    float h_angle, m_angle, angle;
    h = (time[0] - '0')*10 + (time[1] - '0');
    m = (time[3] - '0')*10 + (time[4] - '0');
    h_angle = 0.5 * (60 * h + m);
    m_angle = 6 * m;
    angle = abs(h_angle - m_angle);
    angle = min(360-angle, angle);
    printf("%0.1f", angle);
    return 0;
}